locust/runners.py

```diff
@@ -321,6 +321,7 @@ def stop(self):
     def quit(self):
         for client in self.clients.all:
             self.server.send_to_client(Message("quit", None, client.id))
+        gevent.sleep(0.5) # wait for final stats report from all slaves
```

Comment:
This timing seems kind of arbitrary, is there a reason you went with 0.5?

Reply:
that was my thought also.

Reply:
Yes, it is kind of arbitrary, I'm assuming that the latency between master & slave is < .5s. 

I tried explicitly waiting for one more event, but ran into issues. It could definitely be done better, but I didnt find a way to do it without adding a significant amount of logic.

Reply:
@aldenpeterson-wf @cgoldberg Any more thoughts? The current behaviour is really bad (missed samples on master), so it needs some kind of fix...

Reply:
I guess it could track  which worker nodes have sent their reports, and then some timeout. But I think this is "good enough" and definitely better than the previous behaviour!